app/controllers/pow_controller.rb

```diff
@@ -46,28 +46,45 @@ def index
   def reply
     pow_json = REDIS[server_id].get("POW:#{params[:key]}")
     REDIS[server_id].del("POW:#{params[:key]}")
-    return render plain: "Pow not handed", status: 902 unless pow_json # This pow was never requested or has expired
+    if !pow_json
+      IdentifierService.add_identifier_event(opts, 'Received on Pow Controller, ignored cause Pow not handed')
+      return render plain: "Pow not handed", status: 902 # This pow was never requested or has expired
+    end
     pow = JSON.parse(pow_json)
 
     # dont check for supported clients for now
     # return render plain: "Unsupported data client.", status: 905 unless supported_client?
-    return render plain: "Topic not found.", status: 404 unless TOPICS.include?(params[:topic])
-    return render plain: "Pow not solved correctly", status: 903 unless Digest::SHA2.hexdigest("aod^" + params[:solution] + "^" + params[:key]).unpack("B*")[0].start_with?(pow['wanted'])
-    return render plain: "Payload too large", status: 904 unless params[:natsmsg].bytesize <= NATS_PAYLOAD_MAX
+    if !TOPICS.include?(params[:topic])
+      IdentifierService.add_identifier_event(opts, 'Received on Pow Controller, ignored cause Topic not found')
+      return render plain: "Topic not found.", status: 404
+    end
+
+    if !Digest::SHA2.hexdigest("aod^" + params[:solution] + "^" + params[:key]).unpack("B*")[0].start_with?(pow['wanted'])
+      IdentifierService.add_identifier_event(opts, 'Received on Pow Controller, ignored cause Pow not solved correctly')
+      return render plain: "Pow not solved correctly", status: 903
+    end
+
+    if params[:natsmsg].bytesize > NATS_PAYLOAD_MAX
```

Comment:
this one should have been `if` instead of `unless`... shame on me, nothing to do here :)